using/src/request.rs

```diff
@@ -0,0 +1,62 @@
+//! A `Request` for a spec within a document.
+//!
+//! `Request`s are derived from `Directive`s during schema bootstrapping.
+use graphql_parser::{
+    schema::{Directive, Value},
+    Pos,
+};
+
+use crate::spec::{Spec, SpecParseError};
+
+/// Requests contain a `spec`, the `prefix` requested for that spec (which
+/// will be the spec's [default prefix](Spec.html#default_prefix) if none was
+/// explicitly specified, and the position of the directive making the request
+/// (for validation error reporting).
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Request {
```

Comment:
Wanted to double check that users are expected to use this Struct and it's values? As in: does this need to be `pub`, and do the following three lines need to be `pub`?

Reply:
This question applies to all publicly exposed structs `Spec`, `Implementations`, `Version`, `Pos`, `VersionParseError`. (Definitely see that `Schema` needs to be exposed to the users :)

Reply:
Users are expected to use `Request` (`activations` iterates over `Request`s). I had a general question about whether we should expose the raw fields or make getters. I went with exposing the raw fields over having getters, but could easily be talked into the converse if that's considered poor form.

Reply:
I think the usage of getters is case by case specific: I usually use them for more complicated values that are likely to change drastically, or if I am modelling my API based on a `std` API that already has getters and setters. I think in this case it makes sense to just have them exposed as raw fields!